packages/components/src/validated-form-controls/style.scss

```diff
@@ -4,43 +4,43 @@
 
 .components-validated-control {
 	// For components based on InputBase
-	&:has(:is(input, select):user-invalid)
+	&:has(:is(input, select):invalid[data-validity-visible])
```

Comment:
Changed these selectors from [`:user-invalid`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/:user-invalid) (won't match our "touched" logic exactly) to `:invalid` plus an explicit data attribute that we add based on our own logic of when to start showing the error messages.

Reply:
> Changed these selectors from [`:user-invalid`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/:user-invalid) (won't match our "touched" logic exactly) to `:invalid` plus an explicit data attribute that we add based on our own logic of when to start showing the error messages.

I think I'd missed this part of the current implementation until now. What's different between our logic and the logic of `:user-invalid` ? I quite like the idea if there's any way we could tap into or align with this native browser behavior. It seems pretty close to what we've implemented?